### ISSUE TITLE
Change paragraph definition in rendered `.txt` files

### ISSUE THREAD

**Issue by Seninha** [koreader/koreader#1607]
I read .txt files in Koreader and in the files I write, the paragraphs are separated by blank lines. But koreader renders each line as a single paragraph. So, for get a correct paragraph, I must write a huge line in a plain text that corresponds as a single paragraph, but I don't want to do it, since it looks ugly in my file editor.

E.g., the file abc.txt has the following content:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus.
Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec
consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget
libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut
porta lorem lacinia consectetur.

Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit
amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis,
pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna
consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt
feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec
sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu
rhoncus.
```

How I want it to be rendered:

> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus. Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut porta lorem lacinia consectetur.
> 
> Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis, pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu rhoncus.

How it is rendered:

> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus.<br/>
> Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec<br/>
> consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget<br/>
> libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut<br/>
> porta lorem lacinia consectetur.<br/>
> <br/>
> Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit<br/>
> amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis,<br/>
> pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna<br/>
> consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt<br/>
> feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec<br/>
> sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu<br/>
> rhoncus.<br/>

I.e., the line break keeps unchanged, while I want it to be removed on rendering when there isn't a blank line after it.

**Comment by Frenzie** [comment]
But aside from mysteriously losing the first line, presumably for title purposes, that *is* how it renders?

![screenshot - 08282015 - 09 32 24 am-fs8](https://cloud.githubusercontent.com/assets/202757/9541374/cbdda0a6-4d67-11e5-990d-017dba24f556.png)

It looks like the title only shows up if it's preceded and followed by a blank line.

```

Title

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus.
Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec
consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget
libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut
porta lorem lacinia consectetur.

Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit
amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis,
pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna
consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt
feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec
sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu
rhoncus.
```
![screenshot - 08282015 - 09 36 45 am-fs8](https://cloud.githubusercontent.com/assets/202757/9541433/5a84d1c6-4d68-11e5-9aa7-4ee317e679de.png)


Whereas no blank line following is treated as a paragraph:


```

Title
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus.
Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec
consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget
libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut
porta lorem lacinia consectetur.

Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit
amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis,
pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna
consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt
feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec
sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu
rhoncus.

```
![screenshot - 08282015 - 09 36 27 am-fs8](https://cloud.githubusercontent.com/assets/202757/9541443/68cbc0be-4d68-11e5-855e-006582546690.png)

PS All of this is actually upstream CoolReader stuff.

**Comment by Seninha** [comment]
I got the error. Koreader does not render a chunk of lines as a paragraphs if they are indented. In this case, they remains as single lines and are braked due to the size of the font.

I indent my paragraphs in my `.txt` files, and use the Section names with no indentation. How can I get paragraphs rendered when its lines begins with spaces?

**Comment by Frenzie** [comment]
I don't know to what extent CoolReader can be influenced besides through CSS, but could you perhaps post a working example of the problem?

**Comment by Frenzie** [comment]
The desktop stylesheets seem to have some kind of dynamic variables (see [here](https://github.com/koreader/crengine/blob/c5722dc0c3c517dd8dac15fec2db8a3e60ab1a85/android/res/raw/txt.css#L46)). Adding the `$pre.all` to the stylesheet makes it override stuff like margins so display to some extent makes more sense. This is necessary because of how CR renders this stuff.

Anyway, I've got an initial successful test [here](https://github.com/Frenzie/koreader/commit/cd7ef56f313e283db17256f6757949b160278468). It'll need a little more work before I can make it into a PR though. Note that this effect won't be quite what you're looking for either, unfortunately.

![screenshot - 08292015 - 12 07 30 pm](https://cloud.githubusercontent.com/assets/202757/9561447/8fdeebf2-4e46-11e5-8b14-eeff4e50f668.png)

This screenshot has one tiny difference with the commit I linked above, the font:

```lua
function CreDocument:setTxtPreFormatted(enabled)
    DEBUG("CreDocument: set txt preformatted", enabled)
    self._document:setIntProperty("crengine.file.txt.preformatted", enabled)
    -- monospace is really kind of ugly for this...
    self._document:setStringProperty("styles.pre.font-face", "font-family: \"" .. self.default_font .. "\"")
end
```
